test/powershell/Modules/Microsoft.PowerShell.Core/Pester.Commands.Cmdlets.GetCommand.Tests.ps1

```diff
@@ -49,4 +50,52 @@ Describe "Tests Get-Command with relative paths and wildcards" -Tag "CI" {
         Pop-Location
     }
 
+    It "Get-Command -ShowCommandInfo property field test" {       
+        $properties = ($commandInfo | Get-Member -MemberType NoteProperty)
+        $propertiesAsString =  $properties.name | out-string 
+        $propertiesAsString | Should MatchExactly 'CommandType' 
+        $propertiesAsString | Should MatchExactly 'Definition' 
+        $propertiesAsString | Should MatchExactly 'Module' 
+        $propertiesAsString | Should MatchExactly 'ModuleName' 
+        $propertiesAsString | Should MatchExactly 'Name' 
+        $propertiesAsString | Should MatchExactly 'ParameterSets' 
+    }
+
+    $testcases = @(
+                  @{command = {(Get-Command Get-Date -ShowCommandInfo).Name}; testname = "Name"; result = "Get-Date"}
+                  @{command = {(Get-Command Get-Date -ShowCommandInfo).ModuleName}; testname = "ModuleName"; result = "Microsoft.PowerShell.Utility"}
+                  @{command = {(Get-Command Get-Date -ShowCommandInfo).Module.Name}; testname = "Module"; result = "Microsoft.PowerShell.Utility"}
+                  @{command = {(Get-Command Get-Date -ShowCommandInfo).CommandType}; testname = "CommandType"; result = "Cmdlet"}
+                  @{command = {(Get-Command Get-Date -ShowCommandInfo).Definition.Count}; testname = "Definition"; result = 1}
+               )
+
+    It "Get-Command -ShowCommandInfo property test - <testname>" -TestCases $testcases{
+            param (
+            $command,
+            $result
+        )
+        $command.Invoke() | Should BeExactly $result
+    }
```

Comment:
since you executed `Get-Command Get-Date -ShowCommandInfo` couldn't this just be:
```powershell
$testcases = @{observed = $commandInfo.Name; testname = "Name"; result = "Get-Date"},
    @{observed = $commandInfo.ModuleName; testname = "ModuleName"; result = "Microsoft.PowerShell.Utility"},
    @{observed = $commandInfo.Module.Name; testname = "Module"; result = "Microsoft.PowerShell.Utility"},
    @{observed = $commandInfo.CommandType; testname = "CommandType"; result = "Cmdlet"},
    @{observed = $commandInfo.Definition.Count; testname = "Definition"; result = 1}

    It "Get-Command -ShowCommand property test - <testname>" -testcase $testcases {
        param ( $observed, $result, $testname )
        $observed | Should beExactly $result
    }
```
I'm not sure you need to executed `Get-Command` over and over